api/environments/identities/traits/models.py

```diff
@@ -3,14 +3,31 @@
 from django.core.exceptions import ObjectDoesNotExist
 from django.db import models
 from django.utils.encoding import python_2_unicode_compatible
-from simple_history.models import HistoricalRecords
+from django_lifecycle import AFTER_DELETE, AFTER_SAVE, LifecycleModel, hook
 
 from environments.identities.traits.exceptions import TraitPersistenceError
 from environments.models import BOOLEAN, FLOAT, INTEGER, STRING
 
 
+class TriatQuerySet(models.QuerySet):
```

Comment:
```suggestion
class TraitQuerySet(models.QuerySet):
```

Reply:
Also, what's the benefit of subclassing QuerySet instead of the manager itself?

Reply:
Two reason: 
1. delete is called on queryset
2. We can create can override the bulk_create on the manager but djago itself defines this on Querset
So I think it's more idiomatic, wdyt?

Reply:
You haven't fixed the typo here.